### PR TITLE
UGENE-7891 Add additional filter parameters

### PR DESCRIPTION
**Pull request by DmitriiSukhomlinov**
I will change source branch after [UGENE-7891](https://github.com/ugeneunipro/ugene/pull/1314) will be merged

### CODE REVIEW

File: src/plugins/enzymes/src/EnzymesIO.cpp [hunk 1]
```diff
@@ -121,6 +122,10 @@ QList<SEnzymeData> EnzymesIO::readEnzymes(const QString& url, U2OpStatus& os) {
                     d->overhangTypes |= EnzymeData::OverhangType::Cut3;
                 }
             }
+            //calculate additional parameters
+            d->palindromic = d->seq == DNASequenceUtils::reverseComplement(d->seq, d->alphabet);
+            d->uninterrupted = !d->seq.contains(EnzymeData::UNDEFINED_BASE);
+            d->nondegenerate = d->alphabet->getId() == BaseDNAAlphabetIds::NUCL_DNA_DEFAULT() && !d->seq.contains(EnzymeData::UNDEFINED_BASE);
```
Comment (yalgaer) [inline src/plugins/enzymes/src/EnzymesIO.cpp]: nit: 
`d->nondegenerate = d->alphabet->getId() == BaseDNAAlphabetIds::NUCL_DNA_DEFAULT() && !d->seq.contains(EnzymeData::UNDEFINED_BASE);` 
-> 
`d->nondegenerate = d->alphabet->getId() == BaseDNAAlphabetIds::NUCL_DNA_DEFAULT() && d->uninterrupted;` ?

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp [hunk 2]
```diff
@@ -619,6 +618,9 @@ FindEnzymesDialog::FindEnzymesDialog(ADVSequenceObjectContext* advSequenceContex
     connect(cbMaxLength, QOverload<int>::of(&QComboBox::currentIndexChanged), this, &FindEnzymesDialog::sl_maxLengthChanged);
     connect(enzSel, &EnzymesSelectorWidget::si_newEnzimeFileLoaded, this, &FindEnzymesDialog::sl_updateEnzymesVisibilityWidgets);
     connect(cbOverhangType, QOverload<int>::of(&QComboBox::currentIndexChanged), this, &FindEnzymesDialog::sl_updateVisibleEnzymes);
+    connect(cbShowPalindromic, &QCheckBox::toggled, this, &FindEnzymesDialog::sl_updateVisibleEnzymes);
+    connect(cbShowUninterrupted, &QCheckBox::toggled, this, &FindEnzymesDialog::sl_updateVisibleEnzymes);
+    connect(cbShowNondegenerate, &QCheckBox::toggled, this, &FindEnzymesDialog::sl_updateVisibleEnzymes);
```
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.cpp:623]: `cbShowNondegenerate` automatically includes `cbShowUninterrupted` results.

What do you think if we make `cbShowNondegenerate` enabled & checked only if 'cbShowUninterrupted' is checked?

**Review by yalgaer — COMMENTED** [review]
Looks good.
Will approve after final merges & UI layout check.

File: src/plugins/enzymes/src/EnzymesIO.cpp (hunk 1 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/EnzymesIO.cpp]: OK

File: src/plugins/enzymes/src/FindEnzymesDialog.cpp (hunk 2 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesDialog.cpp:623]: No, my bad, I checked SnapGene again and found out, that "Nondegenerate" enzymes do not contain external DNA alphabet, but could contain Ns. I will rework it.

File: src/plugins/enzymes/src/EnzymesIO.cpp (hunk 1 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/EnzymesIO.cpp]: See comment below

File: src/plugins/enzymes/src/FindEnzymesDialog.ui [hunk 3]
```diff
@@ -270,6 +270,76 @@
...
+         <enum>Qt::Horizontal</enum>
+        </property>
+       </widget>
+      </item>
+      <item>
+       <layout class="QHBoxLayout" name="horizontalLayout_8">
+        <item>
+         <widget class="QLabel" name="label_8">
+          <property name="text">
+           <string>Additional parametes:</string>
```
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesDialog.ui]: Typo in the text

**Review by yalgaer — CHANGES_REQUESTED** [review]
The current UI looks like something is not shown by default and will be added only if the checkbox is checked:
image

I suggest to rework it so a more compact (no repetitive text, move it to the tooltip if needed) and more clear label: 
image

**Comment by DmitriiSukhomlinov** [comment]
Fixed

**Review by yalgaer — APPROVED** [review]

**Pull request merged**